src/org/sosy_lab/java_smt/solvers/princess/PrincessFormulaCreator.java

```diff
@@ -360,8 +363,11 @@ private String getName(IExpression input) {
     }
   }
 
-  /** Returns true if the expression is a constant number. */
-  private static boolean isConstant(IFunApp pExpr) {
+  /** Returns true if the expression is a constant value. */
+  private static boolean isValue(IFunApp pExpr) {
+    if (!pExpr.fun().equals(Rationals.frac()) && !pExpr.fun().equals(Rationals.fromRing())) {
+      return false;
+    }
```

Comment:
Does this work for non-rational numbers? I would assume that Intgers are not longer handled here.